### PR TITLE
Invoke-DbaDatabaseShrink

### PR DESCRIPTION
**Pull request by ctrlbold**
I had to do this for myself, so shared it.

### CODE REVIEW

File: functions/Invoke-DbaDatabaseShrink.ps1 [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+			catch
+			{
+				Stop-Function -Message "Can't connect to $instance. Moving on." -Continue
+			}
+			
+			# changing statement timeout to $StatementTimeout
+			Write-Message -Level Verbose -Message "Changing statement timeout to $StatementTimeout minutes"
+			$server.ConnectionContext.StatementTimeout = $StatementTimeoutMinutes
+			
+			# We have to exclude all the system databases since they cannot have the Query Store feature enabled
```
Comment (niphlod) [inline functions/Invoke-DbaDatabaseShrink.ps1]: leftover note ? querystore shouldn't have to do with shrinking...

File: functions/Invoke-DbaDatabaseShrink.ps1 [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+						
+						$timespan = New-TimeSpan -Start $start -End $end
+						$ts = [timespan]::fromseconds($timespan.TotalSeconds)
+						$elapsed = "{0:HH:mm:ss}" -f ([datetime]$ts.Ticks)
+					}
+				}
+				
+				#$db.TruncateLog()
+				
+				if ($Pscmdlet.ShouldProcess("$db on $instance", "Showing results"))
```
Comment (niphlod) [inline functions/Invoke-DbaDatabaseShrink.ps1]: sure this has to be wrapped in shouldprocess ? it doesn't do anything other than recalculating space usage...

File: functions/Invoke-DbaDatabaseShrink.ps1 (hunk 1 above)
Comment (ctrlbold) [inline functions/Invoke-DbaDatabaseShrink.ps1]: OH, thanks. I did take the querystore stuff.

File: functions/Invoke-DbaDatabaseShrink.ps1 (hunk 2 above)
Comment (ctrlbold) [inline functions/Invoke-DbaDatabaseShrink.ps1]: It recalcs and also outputs the object. Plus, a recalculate is a write so I'd like to protect that.

**Pull request merged**